src/sciline/pipeline.py

```diff
@@ -793,7 +752,13 @@ def compute(self, tp: Iterable[Type[T]], **kwargs: Any) -> Dict[Type[T], T]:
     def compute(self, tp: Item[T], **kwargs: Any) -> T:
         ...
 
-    def compute(self, tp: type | Iterable[type] | Item[T], **kwargs: Any) -> Any:
+    @overload
+    def compute(self, tp: object, **kwargs: Any) -> Any:
```

Comment:
Similar to above, this is equivalent to 
```suggestion
    def compute(self, tp: Any, **kwargs: Any) -> Any:
```
if I'm not mistaken.
Does mypy use the specialised overloads where possible or does it always fall back to `Any`?